cpp/src/io/csv/csv_writer.cu

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+     * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file csv_writer.cu  code to create csv file
+ *
+ * CSV Writer
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <cuda_runtime.h>
+#include <nvstrings/NVStrings.h>
+#include <rmm/rmm.h>
+#include <rmm/thrust_rmm_allocator.h>
+#include "cudf.h"
+#include "utilities/error_utils.hpp"
+
+
+// called by write_csv method
+NVStrings* column_to_strings_csv( gdf_column* col, const char* delimiter, const char* strue, const char* sfalse )
+{
+    NVStrings* rtn = nullptr;
+    unsigned int rows = (unsigned int)col->size;
+    unsigned char* valid = (unsigned char*)col->valid;
+    switch( col->dtype )
+    {
+        case GDF_STRING:
+            rtn = ((NVStrings*)col->data)->copy();
+            break;
+        case GDF_INT8:
+            rtn = NVStrings::create_from_bools((const bool*)col->data,rows,strue,sfalse,valid);
+            break;
+        case GDF_INT32:
+            rtn = NVStrings::itos((const int*)col->data,rows,valid);
+            break;
+        case GDF_INT64:
+            rtn = NVStrings::ltos((const long*)col->data,rows,valid);
+            break;
+        case GDF_FLOAT32:
+            rtn = NVStrings::ftos((const float*)col->data,rows,valid);
+            break;
+        case GDF_FLOAT64:
+            rtn = NVStrings::dtos((const double*)col->data,rows,valid);
+            break;
+        case GDF_DATE64:
+            rtn = NVStrings::long2timestamp((const unsigned long*)col->data,rows,NVStrings::seconds,valid);
+            break;
+        default:
+            break; // should not happen
```

Comment:
But it will happen, because we support other data types than those listed. What happens if the type is GDF_DATE32, GDF_CATEGORY, GDF_STRING_CATEGORY, etc? 

Also, this is why we created `type_dispatcher` -- to avoid having hundreds of switches over gdf_dtype. See https://github.com/rapidsai/cudf/blob/997aa847d467bf4a65bb9a03aef82a57656bf4fc/cpp/src/utilities/type_dispatcher.hpp  It won't work as-is since every type has a different conversion function, but I'm sure @jrhemstad has opinions on the right way to handle it.

Reply:
The caller of this method ensures only these types are expected by this method. I could remove the comment if that helps.

Reply:
Would like to leave this as switch (of if-else pattern) for this PR. Not all the types have been accounted for but Randy wanted at least an initial version to make some progress. I'd rather wait until we have all the types figured out before engaging the type-dispatcher here. There will be future PRs for this to meet up with Randy's requirements.

Reply:
In order to better work with the `type_dispatcher`, it would make more sense for `NVStrings` to have a function template for converting a column of any type `T` to an `NVStrings`, rather than having separate functions for each possible column type.

In the meantime, in this code, at the very least you should throw an exception if the default case is reached. `CUDF_FAIL` would be appropriate here: https://github.com/rapidsai/cudf/blob/branch-0.7/cpp/src/utilities/error_utils.hpp#L123